### PR TITLE
vgpu feature for volcano(device-plugin part)

### PR DESCRIPTION
**Pull request by archlitchi**
Add vgpu feature for volcano (device-plugin part)

### CODE REVIEW

File: cmd/vgpu4pd/main.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2020 The Volcano Authors.
```
Comment (william-wang) [inline cmd/vgpu4pd/main.go]: please change the directory name to `cmd/vgpu/main.go`

File: cmd/vgpu4pd/main.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+	nvidiadevice "volcano.sh/k8s-device-plugin/pkg/plugin/vgpu4pd"
+	"volcano.sh/k8s-device-plugin/pkg/plugin/vgpu4pd/config"
+	"volcano.sh/k8s-device-plugin/pkg/plugin/vgpu4pd/util"
+)
+
+var (
+	failOnInitErrorFlag bool
+	//nvidiaDriverRootFlag string
+	//enableLegacyPreferredFlag bool
```
Comment (william-wang) [inline cmd/vgpu4pd/main.go]: Suggest remove the useless comments.

File: cmd/vgpu4pd/watchers.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2020 The Volcano Authors.
```
Comment (william-wang) [inline cmd/vgpu4pd/watchers.go]: Please change the Copyright time.

File: pkg/plugin/vgpu4pd/cache.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
+/*
```
Comment (william-wang) [inline pkg/plugin/vgpu4pd/cache.go]: Change the directory name to `pkg/plugin/vgpu/cache.go`

File: pkg/plugin/vgpu4pd/mig-strategy.go [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+	"github.com/NVIDIA/gpu-monitoring-tools/bindings/go/nvml"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+	"volcano.sh/k8s-device-plugin/pkg/plugin/vgpu4pd/util"
+)
+
+// Constants representing the various MIG strategies
+const (
+	MigStrategyNone   = "none"
+	MigStrategySingle = "single"
+	MigStrategyMixed  = "mixed"
```
Comment (william-wang) [inline pkg/plugin/vgpu4pd/mig-strategy.go]: What's the MigStragey used for and it's difference with nvidia MIG.

File: pkg/plugin/vgpu4pd/mig-strategy.go [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+}
+
+func (s *migStrategyNone) MatchesResource(mig *nvml.Device, resource string) bool {
+	panic("Should never be called")
+}
+
+// migStrategySingle
+func (s *migStrategySingle) GetPlugins(cache *DeviceCache) []*NvidiaDevicePlugin {
+	panic("single mode in MIG currently not supported")
+	/*devices := NewMIGCapableDevices()
```
Comment (william-wang) [inline pkg/plugin/vgpu4pd/mig-strategy.go]: Suggest remove the useless comments codes.

**Review by william-wang — APPROVED** [review]
/lgtm
/approve

**Comment by wangyang0616** [comment]
The code has been inspected ok, and marked with `/lgtm` and `/approve` flags, but the flags are not automatically synchronized to the label of the PR, it seems that prowbot has not taken effect

**Comment by wangyang0616** [comment]
tide/merge-method-merge

**Comment by wangyang0616** [comment]
/tide/merge-method-merge

**Pull request merged**